### ISSUE TITLE
It's different with the standard SegNet

### ISSUE THREAD

**Issue by deeptoby** [bodokaiser/piwise#24]
I think the most contribution in segnet is the idx-maxpooling. you can use the  F.max_unpool2d(idx, x) to replace the nn.upsample().
The original upsample can't fit the loss of location information.

**Comment by bodokaiser** [comment]
Hey, yes that makes sense! Would you mind to send a PR?

**Comment by deeptoby** [comment]
yes, i'd like to do that. i have rewrited segnet.i will send it by my pull.

**Comment by Viarow** [comment]
self.dec1 = features[0: 4]    #[0,3]
        self.dec2 = features[5: 9]    #[5,8]
        self.dec3 = features[10: 16]  #[10,15]
        self.dec4 = features[17: 23]  #[17,22]
        self.dec5 = features[24: -1]  #[24,29]
while there's also a maxpool2d layer after each decoder block in the forward function, I wonder why the pooling operation is implemented twice before unpooling? I haven't run the code just quite confused.

**Comment by bodokaiser** [comment]
@Viarow I guess this is a mistake then. Would you mind trying the correct version?

**Comment by Viarow** [comment]
I'm actually a novice at deep learning, still trying to understand the codes. I guess it will work right only if the range of pretrained layers are set correct (as shown in the annotations above).  I will tell you the results after I try the corrections.

**Comment by deeptoby** [comment]
Hi,there just one pooling at each dec.  could you tell me the your found on detail.

**Comment by bodokaiser** [comment]
If I understood correctly the problem can be found with the index selectors here:
    self.dec1 = features[0: 4]    #[0,3]
    self.dec2 = features[5: 9]    #[5,8]
    self.dec3 = features[10: 16]  #[10,15]
    self.dec4 = features[17: 23]  #[17,22]
    self.dec5 = features[24: -1]  #[24,29]

So for example [0:4] will return [features[0], features[1], features[3]].

As the next layer starts with features[5] the features[4] layer is not used at all.

**Comment by deeptoby** [comment]
layer4 is maxpool, it is used later in code in order to get max-index.

**Comment by Viarow** [comment]
Oh I see the point. in the FCN8 model,
        self.feats = nn.Sequential(*feats[0:9])
        self.feat3 = nn.Sequential(*feats[10:16])
        self.feat4 = nn.Sequential(*feats[17:23])
        self.feat5 = nn.Sequential(*feats[24:30])
are called later in 
        feats = self.feats(x)
        feat3 = self.feat3(feats)
        feat4 = self.feat4(feat3)
        feat5 = self.feat5(feat4)
        fconn = self.fconn(feat5)
which means that some maxpooling layers are also left out in this model?
